### DISCUSSION TITLE
В Alpine Linux установлено, но не работает

### DISCUSSION THREAD

**Discussion by Ojlerceu4** [xvzc/SpoofDPI#160] [Q&A] [answered]
Здравствуйте.
Alpine linux установлен гостем в proxmox. Архитектура amd64. SpoofDPI установлено скриптом, запускаемый файл появился, но при запуске программа пишет что не находит требуемый файл - очевидно что spoof-dpi запускается, но ему требуется ещё что-то отсутствующее - вероятно в Alpine надо что-то еще установить?

# spoof-dpi
-bash: /root/.spoof-dpi/bin/spoof-dpi: cannot execute: required file not found

# ls -l /root/.spoof-dpi/bin/
total 6356
-rwxr-xr-x    1 1001     127        6508544 Aug 15 04:33 spoof-dpi

**Comment by asimba** [comment]
Ключ -system-proxy не пробовали отключать?

**Reply by Ojlerceu4** [reply]
Результат тот-же:
# spoof-dpi -system-proxy
-bash: /root/.spoof-dpi/bin/spoof-dpi: cannot execute: required file not found

и с  -system-proxy=false то же, даже просто -h вместо вывода подсказок выдаёт то же самое

**Reply by asimba** [reply]
Может проще из сырцов пересобрать?

**Reply by asimba** [reply]
Кстати, немного оффтопа. Судя по вашей установке, возможен сценарий использования SpoofDPI в локальной сети в качестве прокси для нескольких машин. Если я прав, то обратите внимание: прокси SpoofDPI на локальном интерфейсе вместо 127.0.0.1, во всяком случае у меня, периодически "терял пакеты", а вот когда поверх поставил Privoxy с перенаправлением на 127.0.0.1 всё заработало как надо. Бонусом через тот же privoxy можно настроить проброс в tor и/или i2p (если они установлены), ну и фильтрация более гибкая.

**Reply by Ojlerceu4** [reply]
Именно так и сделал, потому и в виртуалке. Пока не смогу оживить на alpine - работает на дебиане и доступен для всех моих устройств в локалке. Работает.
Надеюсь что удастся выяснить причины неработоспособности на alpine и задействовать с ним, чтобы расход ресурсов серверка на это резко сократить.
С дебиан занятого места чуть меньше 2ГБ, опеативки, без учета кэша, занимает около 90МБ, а выделять приходится минмум 300МБ, иначе дебиан отказывается грузиться.
С альпин - занятого места меньше 100МБ, а оперативной занято меньше 40, выделено 128.
Разница ощутимая - надеюсь что удастся разобраться с работой SpoofDPI на Alpine.

Автора благодарю! Помог вернуть возможность просмотра обучалок, обзоров и прочего.

**Reply by asimba** [reply]
На отдельном Alpine попробуйте пересобрать бинарник с установленной переменной среды CGO_ENABLED=0 и ключом -ldflags="-s -w". Может такой заработает.

**Comment by 00enterausername00** [comment]
У меня в докере завелось двумя командами:
```
apk add --no-cache curl bash
curl -fsSL https://raw.githubusercontent.com/xvzc/SpoofDPI/main/install.sh | bash -s linux-arm64
```
в твоем случае вторую команду только под другую архитектуру нужно использовать

**Reply by Ojlerceu4** [reply, marked as answer]
Да, разумеется так  и савил - скрипт получаемый и запускаемый этой строкой скачивает последний архив с запускаемым файлом, помещает его в каталог пользователя и добавляет путь к нему путь туда.
Вероятно,  в alpine чего-то не хватает или присутствует не в таком виде как надо для работы файла spoof-dpi - выяснить бы что именно дбавить или изменить, если возможно.

Заменил на 0.10.7 - работает. Вопрос, думаю, можно считать решённым.
